src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -105,19 +109,22 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
             } elseif ($argument instanceof MapRequestPayload) {
                 $payloadMapper = 'mapRequestPayload';
                 $validationFailedCode = $argument->validationFailedStatusCode;
+            } elseif ($argument instanceof MapUploadedFile) {
+                $payloadMapper = 'mapUploadedFile';
```

Comment:
Note, which need not be included in this PR: First Class Callable should be used here.
```diff
- $payloadMapper = 'mapUploadedFile';
+ $payloadMapper = $this->mapUploadedFile(...);

// ...

- $payload = $this->$payloadMapper($request, $argument->metadata, $argument);
+ $payload = $payloadMapper($request, $argument->metadata, $argument);
```